### ISSUE TITLE
Regression in loading variables.tf.json from module

### ISSUE THREAD

**Issue by phinze** [hashicorp/terraform#8820]
### Terraform Version

`0.7.3`

### Terraform Configuration Files

```hcl
module "ami" {
  source        = "github.com/terraform-community-modules/tf_aws_ubuntu_ami/ebs"
  instance_type = "t2.micro"
  region        = "us-east-1"
  distribution  = "trusty"
}

output "ami" {
  value = "${module.ami.ami_id}"
}

```

### Steps to Reproduce

1. `terraform get`

### Expected Behavior

* Successful exit

### Actual Behavior

```
Get: git::https://github.com/terraform-community-modules/tf_aws_ubuntu_ami.git
Get: git::https://github.com/terraform-community-modules/tf_aws_virttype.git
Get: file:///Users/phinze/tmp/tf/amimoderror/.terraform/modules/0ec790a5c99e8db6524c7ac02e48e712
Error loading Terraform: module ami.root: module virttype: required variable prefer_pv not set
```

### Important Factoids

 * The submodule expects `variables.tf.json` to define the default value for `prefer_pv`
 * Works fine on Terraform 0.7.2
 * Running master w/ fixes in #8803 do not resolve the issue.

### References

 * #8398 - Instigating change
 * #8724 - Related bug
 * #8803 - Related but insufficient bugfix

**Comment by jbardin** [comment]
The breaking commit turns out to be #8591

**Comment by jbardin** [comment]
The root cause is a bug in HCL: https://github.com/hashicorp/hcl/issues/153

**Comment by jbiel** [comment]
Thanks for the fix, this hit me on 0.7.3. When I passed the config file via -var-file... I still ended up with some kind of parsing issue because I received an error ("unknown variable accessed".) Master works (but I keep trying to get away from running it!)
